include/up-cpp/transport/datamodel/UPayload.h

```diff
@@ -39,124 +39,118 @@ namespace uprotocol::utransport {
         UNDEFINED  /* invalid */
     };
 
+    // The Serialization format for the data stored in UPayload.
+    enum UPayloadFormat {
+        UNSPECIFIED = 0,             /* Payload format was not is not set */
+        PROTOBUF_WRAPPED_IN_ANY = 1, /* Payload is an Any protobuf message that contains the packed payload */
+        PROTOBUF = 2,                /* Protobuf serialization format */
+        JSON = 3,                    /* JSON serialization format */
+        SOMEIP = 4,                  /* Basic SOME/IP serialization format */
+        SOMEIP_TLV = 5,              /* SOME/IP TLV format */
+        RAW = 6,                     /* RAW (binary) format */
+        TEXT = 7                     /* Text format */
+    };          
+
     /**
     * The UPayload contains the clean Payload information at its raw serialized structure of a byte[]
     */
     class UPayload {
-        public:
 
-            UPayload(const uint8_t *data, 
-                     const size_t &dataSize,
-                     const UPayloadType &type) {
-                
-                payloadType_ = type;
+        public:
+            // Constructor
+            UPayload(const uint8_t* ptr, 
+                     const size_t size, 
+                     const UPayloadType &type) : dataSize_(size), type_(type) {
 
-                if ((nullptr == data) || (0 == dataSize)) {
-                    data_ = nullptr;
-                    refPtr_ = nullptr;
-                    dataSize_ = 0;
+                if (type == UPayloadType::REFERENCE) {
+                    dataPtr_ = std::shared_ptr<const uint8_t[]>(ptr, [](const uint8_t*){});
+                } else {
+                    dataPtr_ = std::shared_ptr<const uint8_t[]>(new uint8_t[dataSize_], [](const uint8_t* p){ delete[] p; });
```

Comment:
Consider using `std::make_shared<std::vector>(dataSize_)` instead

(NB: C++20 would allow the use of `std::make_shared<const uint8_t[]>(dataSize_)`)